Prevent PTable from interfering with prettytable.

jdblischak opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [ ] Ensured the license file is being packaged.

I have submitted the PyPI package [PTable](https://pypi.org/project/PTable/) to staged-recipes (https://github.com/conda-forge/staged-recipes/pull/7127). It is a fork of prettytable, and thus could interfere with it. This PR prevents PTable and prettytable from being installed in the same environment.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jdblischak commented:
The [import test failed](https://circleci.com/gh/conda-forge/prettytable-feedstock/31):

```
TEST START: /home/conda/feedstock_root/build_artifacts/noarch/prettytable-0.7.2-py_3.tar.bz2
Renaming work directory,  /home/conda/feedstock_root/build_artifacts/prettytable_1543876728639/work  to  /home/conda/feedstock_root/build_artifacts/prettytable_1543876728639/work_moved_prettytable-0.7.2-py_3_noarch
import: 'prettytable'
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/prettytable_1543876728639/test_tmp/run_test.py", line 2, in <module>
    import prettytable
ModuleNotFoundError: No module named 'prettytable'
Tests failed for prettytable-0.7.2-py_3.tar.bz2 - moving package to /home/conda/feedstock_root/build_artifacts/broken
```

jdblischak commented:
I restarted the build by force pushing an amended commit. Maybe now that [ptable is from conda-forge](https://anaconda.org/conda-forge/ptable) and conda 3.17.1 has been released (with some bug fixes), this could resolve the issue.

jdblischak commented:
The build passed! @conda-forge/prettytable this PR is ready for review

jdblischak commented:
@scopatz Do you have permission to merge this PR?

scopatz commented:
Thanks @jdblischak

jdblischak commented:
Thanks @scopatz! I confirmed it works as expected:

```
$ conda create -y -n test-pt prettytable
Solving environment: done

## Package Plan ##

  environment location: /home/jdb-work/miniconda3-conda-forge/envs/test-pt

  added / updated specs: 
    - prettytable


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    certifi-2018.11.29         |        py37_1000         145 KB  conda-forge
    ncurses-6.1                |       hfc679d8_2         1.3 MB  conda-forge
    ------------------------------------------------------------
                                           Total:         1.4 MB

The following NEW packages will be INSTALLED:

    bzip2:           1.0.6-h470a237_2      conda-forge
    ca-certificates: 2018.11.29-ha4d7672_0 conda-forge
    certifi:         2018.11.29-py37_1000  conda-forge
    libffi:          3.2.1-hfc679d8_5      conda-forge
    libgcc-ng:       7.2.0-hdf63c60_3      conda-forge
    libstdcxx-ng:    7.2.0-hdf63c60_3      conda-forge
    ncurses:         6.1-hfc679d8_2        conda-forge
    openssl:         1.0.2p-h470a237_1     conda-forge
    pip:             18.1-py37_1000        conda-forge
    prettytable:     0.7.2-py_2            conda-forge
    python:          3.7.1-h5001a0f_0      conda-forge
    readline:        7.0-haf1bffa_1        conda-forge
    setuptools:      40.6.2-py37_0         conda-forge
    sqlite:          3.26.0-hb1c47c0_0     conda-forge
    tk:              8.6.9-ha92aebf_0      conda-forge
    wheel:           0.32.3-py37_0         conda-forge
    xz:              5.2.4-h470a237_1      conda-forge
    zlib:            1.2.11-h470a237_3     conda-forge


Downloading and Extracting Packages
certifi-2018.11.29   | 145 KB    | ############################### | 100% 
ncurses-6.1          | 1.3 MB    | ############################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use:
# > source activate test-pt
#
# To deactivate an active environment, use:
# > source deactivate
#

$ source activate test-pt
(test-pt) $ conda install -y ptable
Solving environment: done

## Package Plan ##

  environment location: /home/jdb-work/miniconda3-conda-forge/envs/test-pt

  added / updated specs: 
    - ptable


The following NEW packages will be INSTALLED:

    ptable:      0.9.2-py_0 conda-forge

The following packages will be REMOVED:

    prettytable: 0.7.2-py_2 conda-forge

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

```
